MaterialBudget.h

```diff
@@ -3,27 +3,44 @@
 
 #include<iostream>
 #include"TRandom3.h"
-#include"Particle.h"
+#include "TFile.h"
+#include "TTree.h" 
+#include "TBranch.h"
+#include "Particle.h"
 
 class MaterialBudget : public TObject
 {
 
 public:
+
+    using fPoint = struct{
+        bool isIntersection=false; double x = 0, y = 0, z = 0, phi = 0;     //no need for theta in cilindrical coordinates
+        void print() {cout<<"isIntersection: "<<isIntersection<<"\tx: "<<x<<"\ty: "<<y<<"\tz: "<<z<<"\tphi: "<<phi<<endl;}
+    };
+
     MaterialBudget();                                                       // default constructor
-    MaterialBudget(double height, double radius, double length, double density, int z, int a);
-    MaterialBudget(double height, double radius, double length, string material);
+    MaterialBudget(double height, double radius, double length, double density, int z, int a, bool multscat);
+    MaterialBudget(double height, double radius, double length, bool multscat, string material);
     
     double GetHeight() const                               {return fHeight;}
     double GetRadius() const                               {return fRadius;}
     double GetLength() const                               {return fLength;}          
     vector<double> GetGeometry() const                     {return {fHeight,fRadius,fLength};}
     double GetDensity() const                              {return fDensity;}
     vector<int> GetFeatures() const                        {return {fZ, fA};}
+    bool GetStatus() const                                 {return fMultScat;}
+
+    virtual fPoint GetIntersection(const Particle* particle, bool);
 
+    void SetStatus(bool multscat);
```

Comment:
I would add an overolad funcion which takes a vector<bool> as parameter (same for the set status of detector)